plugins/services/src/js/components/ServiceStatusWarning.js

```diff
@@ -7,29 +7,48 @@ import Icon from '../../../../../src/js/components/Icon';
 import Pod from '../structs/Pod';
 import Service from '../structs/Service';
 import ServiceTree from '../structs/ServiceTree';
+import StringUtil from '../../../../../src/js/utils/StringUtil';
 
 class ServiceStatusWarning extends Component {
+  getTooltip(content) {
+    return (
+      <Tooltip
+        content={content}
+        width={250}
+        wrapText={true}
+        wrapperClassName="tooltip-wrapper status-waiting-indicator">
+        <Icon color="red" id="yield" size="mini" />
+      </Tooltip>
+    );
+  }
+
   getTooltipContent(timeWaiting) {
     return `DC/OS has been waiting for resources and is unable to complete this deployment for ${DateUtil.getDuration(timeWaiting, null)}.`;
   }
+
   render() {
     const {item} = this.props;
-    const queue = item.getQueue();
-
-    if (queue != null
-      && DeclinedOffersUtil.shouldDisplayDeclinedOffersWarning(queue)) {
-      const waitingSince = DateUtil.strToMs(queue.since);
-      const timeWaiting = Date.now() - waitingSince;
-
-      return (
-        <Tooltip
-          content={this.getTooltipContent(timeWaiting)}
-          width={250}
-          wrapText={true}
-          wrapperClassName="tooltip-wrapper status-waiting-indicator">
-          <Icon color="red" id="yield" size="mini" />
-        </Tooltip>
-      );
+
+    if (item instanceof ServiceTree) {
```

Comment:
Alright, but could we at least get rid of that `getTooltipContent` function, that doesn't really do anything for us we could not just do in this function?